File: fix_duplicates.dart
import 'dart:io';

void main(List<String> args) {
  if (args.isEmpty) return;
  final file = File(args.first);
  final content = file.readAsStringSync();
  final lines = content.split('\n');
  final newLines = <String>[];
  
  var inMap = false;
  var currentKeys = <String>{};
  
  final mapStartRegExp = RegExp(r"^('.+'):\s*\{");
  final constMapRegExp = RegExp(r"^const\s+_?[a-zA-Z0-9]+\s*=\s*<String,\s*String>\{");
  final keyRegExp = RegExp(r"^('[^']+')\s*:");

  for (final line in lines) {
    final stripped = line.trim();
    if (mapStartRegExp.hasMatch(stripped) || constMapRegExp.hasMatch(stripped)) {
      inMap = true;
      currentKeys.clear();
      newLines.add(line);
      continue;
    }
    if (inMap && (stripped.startsWith('}') || stripped == '};')) {
      inMap = false;
      newLines.add(line);
      continue;
    }
    if (inMap) {
      final match = keyRegExp.firstMatch(stripped);
      if (match != null) {
        final key = match.group(1)!;
        if (currentKeys.contains(key)) {
          print('Removing duplicate key $key');
          continue;
        } else {
          currentKeys.add(key);
        }
      }
    }
    newLines.add(line);
  }
  
  file.writeAsStringSync(newLines.join('\n'));
  print('Done.');
}

File: fix_duplicates.py
import re
import sys

def fix_duplicates(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    lines = content.split('\n')
    new_lines = []
    
    in_map = False
    current_keys = set()
    
    for line in lines:
        stripped = line.strip()
        
        # Matches either "'en': {" OR "const _en = <String, String>{"
        if (stripped.startswith("'") and "': {" in stripped) or (stripped.startswith("const") and "<String, String>{" in stripped):
            in_map = True
            current_keys = set()
            new_lines.append(line)
            continue
            
        if in_map and (stripped.startswith("}") or stripped == "};"):
            in_map = False
            new_lines.append(line)
            continue
            
        if in_map:
            # Check if it's a key-value pair
            match = re.match(r"^('[^']+')\s*:", stripped)
            if match:
                key = match.group(1)
                if key in current_keys:
                    print(f"Removing duplicate key {key} in {filepath}")
                    continue # Skip this line
                else:
                    current_keys.add(key)
        
        new_lines.append(line)

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write('\n'.join(new_lines))

fix_duplicates(sys.argv[1])
